src/Common/Internal/Http/HttpClient.php

```diff
@@ -385,3 +399,6 @@ public static function throwIfError($actual, $reason, $message, $expected)
         }
     }
 }
+
+// read HTTP_PROXY enviroment variable, if any
+HttpClient::initProxy();
```

Comment:
Agree. Usually, a singleton is anti-pattern. Replaced with local parsing of the environment variable.
https://github.com/Azure/azure-sdk-for-php/pull/864/files#diff-e16bf3a450a1c82ec99cfa0c730be802R92